application/client.plugins/projects/serial/src/lib/services/service.ts

```diff
@@ -87,7 +88,7 @@ export class Service extends Toolkit.APluginService {
                     this.state = response;
                     this._subjects.event.next(message);
                 }).catch((error: Error) => {
-                    this._logger.error(error);
+                    throw(error);
```

Comment:
using `throw` will drop an exception and break the whole JS thread. In addition, because you are calling `throw` from a promise it will be uncatched promise exception.

Why we need throw here?

Reply:
Also in many other places of code you have `throw`. We should get rid of it or if it's really needed wrap it into `try {} catch {}` to prevent breaking of the whole workflow

Reply:
I would do more. Because it's front-end (render), the plugin process is included in the main core JS thread. It means `throw` will break also the core process. So, I would add a strict rule for plugin's developing - using of unwrapped throws are prohibited at all on render side. 

So it could be reflected in documentations.